Cisco/cisco-ftd/ingest/parser.yml

```
@@ -0,0 +1,210 @@
+name: cisco-ftd
+pipeline:
+  - name: pre_parsing
+    external:
+      name: grok.match
+      properties:
+        input_field: "original.message"
+        output_field: pre_message
+        pattern: '\%(%{FTD:product}|%{ASA:product}|%{NGIPS:product})\-[0-9]{1}\-%{MESSAGE_NUMBER:message_number_grok}\: %{DATA:raw_message}'
+        custom_patterns: 
+          MESSAGE_NUMBER: '[0-9]{6}'
+          FTD: 'FTD'
+          ASA: 'ASA'
+          NGIPS: 'NGIPS'
```

I wonder if you can't optimize with:
```suggestion
        pattern: '\%%{PRODUCT_NAME:product}\-[0-9]{1}\-%{MESSAGE_NUMBER:message_number_grok}\: %{DATA:raw_message}'
        custom_patterns: 
          MESSAGE_NUMBER: '[0-9]{6}'
          PRODUCT_NAME: '(FTD|ASA|NGIPS)'
```